src/arrayMethodJoin.js

```diff
@@ -4,8 +4,28 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let outString = '';
+
+    for (let i = 0; i < this.length; i++) {
+      switch (this[i]) {
+        case undefined:
+          outString += '';
+          break;
+        case null:
+          outString += '';
+          break;
+        default:
+          outString += this[i];
+          break;
+      }
```

Comment:
Looks too complicated and could be replaced with:
`if (this[i] !== null && this[i] !== undefined) {
  outString += this[i];
}`

Reply:
Not fixed